src/EventTickets/ServiceProvider.php

```diff
@@ -90,6 +91,7 @@ private function registerEventTicketsAdminPage(): void
     {
         Hooks::addAction('admin_menu', RegisterEventsMenuItem::class, '__invoke', 15);
         Hooks::addAction('admin_enqueue_scripts', EnqueueListTableScripts::class);
+        Hooks::addAction('admin_enqueue_scripts', EnqueueEventDetailsScripts::class);
```

Comment:
What if we called the enqueue actions from within the render method of the submenu page registration? Then we wouldn't need the `isShowing()` check and the logic of which script gets enqueued is co-located with the page render.

```php
class RegisterEventsMenuItem
{
    public function render()
    {
        get_query_var('id')
            ? give(EnqueueEventDetailsScripts::class)()
            : give(EnqueueListTableScripts::class)();
        
        echo '<div id="give-admin-event-tickets-root"></div>';
    }
}
```

Reply:
Here is a quick PR showing / explaining my suggestion, see https://github.com/impress-org/givewp/pull/7262.